### PR TITLE
Compressible DG cylindrical test

### PR DESCRIPTION
**Pull request by drhodrid**
Addition of a Compressible (TALA) DG cylindrical test, to showcase our DG capabilities, and also catch any potential issues with our DG code. Note that this is reliant on https://github.com/g-adopt/g-adopt/pull/98.

Expected values for new demo regression testing as follows:

1757 0.9026195309711268 0.0005164681112863269 9.971121789179138e-08 37.47725190694167 3.5080693782419017 3.5096072680824646 0.0015378898405629116 0.4273985482694366 0.0038372777142413623 0.9985517674877391 1.7056159827176107 1.7486780146166836

### CODE REVIEW

**Comment by drhodrid** [comment]
Note - no review necessary at this stage. I still have some text to improve in the demo.

**Comment by angus-g** [comment]
> Note - no review necessary at this stage. I still have some text to improve in the demo.

Do you want to make it a draft then?

**Comment by drhodrid** [comment]
OK. I have moved this across from a demo to the test, and communicated with @angus-g on the required params and initial condition. The test is our first of DG, is in a cylinder, and verifies that the max temperature in the domain doesn't diverge too much from the prescribed boundary conditions. Prior to https://github.com/g-adopt/g-adopt/pull/98, this was not the case.

File: tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+cpbar = Function(Q, name="IsobaricSpecificHeatCapacity")
+interpolate_1d_profile(function=cpbar, one_d_filename="initial_condition_mat_prop/CpSIbar.txt")
+cpbar.assign(cpbar / 1249.7)
+gbar = Function(Q, name="GravitationalAcceleration")
+interpolate_1d_profile(function=gbar, one_d_filename="initial_condition_mat_prop/gbar.txt")
+gbar.assign(gbar / 9.8267)
+
+# We next prepare our viscosity, starting with a radial profile.
+mu_rad = Function(Q, name="Viscosity_Radial")  # Depth dependent component of viscosity
+interpolate_1d_profile(function=mu_rad, one_d_filename="initial_condition_mat_prop/mu2_radial.txt")
```
Comment (angus-g) [inline tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py:91]: None of these profiles are included in this PR

File: tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+# Given the increased computational expense (typically requiring more degrees of freedom) in a 2-D annulus domain, G-ADOPT defaults to iterative
+# solver parameters. As noted in our previous 3-D Cartesian tutorial, G-ADOPT's iterative solver setup is configured to use the GAMG preconditioner
+# for the velocity block of the Stokes system, to which we must provide near-nullspace information, which, in 2-D, consists of two rotational and two
+# translational modes.
+
+Z_near_nullspace = create_stokes_nullspace(Z, closed=False, rotational=True, translations=[0, 1])
+
+# Boundary conditions are next specified. Boundary conditions for temperature are set to $T = 0$ at the surface ($r_{\text{max}}$) and $T = 1 - adiabatic contribution$
+# at the base ($r_{\text{min}}$). For velocity, we specify free‐slip conditions on both boundaries. We incorporate these <b>weakly</b> through
+# the <i>Nitsche</i> approximation. This illustrates a key advantage of the G-ADOPT framework: the user only specifies that the normal component
```
Comment (angus-g) [inline tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py:142]:
```suggestion
# at the base ($r_{\text{min}}$). For velocity, we specify free‐slip conditions on both boundaries. We incorporate these **weakly** through
# the _Nitsche_ approximation. This illustrates a key advantage of the G-ADOPT framework: the user only specifies that the normal component
```

File: tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+# We next setup our output, in VTK format.
+# We also open a file for logging and calculate our diagnostic outputs.
+
+# +
+output_file = VTKFile("output.pvd")
+ref_file = VTKFile('reference_state.pvd')
+output_frequency = 10
+
+plog = ParameterLog('params.log', mesh)
```
Comment (angus-g) [inline tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py:165]:
```suggestion
ref_file = VTKFile("reference_state.pvd")
output_frequency = 10

plog = ParameterLog("params.log", mesh)
```

File: tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py (hunk 1 above)
Comment (drhodrid) [reply tests/2d_cylindrical_TALA_DG/2d_cylindrical_TALA_DG.py:91]: Added.

**Comment by drhodrid** [comment]
Files added, and review suggestions incorporated. Thanks Angus!

**Comment by angus-g** [comment]
Which of the logged parameters (u_rms nu_base nu_top energy avg_t FullT_min FullT_max) do you want to test, noting that we usually only test u_rms and nu_top, except for 2d_cylindrical which adds T_min and T_max?

**Comment by drhodrid** [comment]
> Which of the logged parameters (u_rms nu_base nu_top energy avg_t FullT_min FullT_max) do you want to test, noting that we usually only test u_rms and nu_top, except for 2d_cylindrical which adds T_min and T_max?

Is it possible to test them all? The latter two are particularly sensitive if we get something wrong in DG, so definitely need them tested!!

**Review by angus-g — APPROVED** [review]
